hivemind/utils/crypto.py

```diff
@@ -0,0 +1,84 @@
+import re
+from dataclasses import astuple, dataclass
+from typing import Optional
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import padding, rsa
+
+from hivemind.utils.serializer import MSGPackSerializer
+
+
+@dataclass(init=True, repr=True, frozen=True)
+class ProtectedRecord:
+    key: bytes
+    subkey: bytes
+    value: bytes
+    expiration_time: float
```

Comment:
Question: do we need to add sender endpoint here?

Reply:
I don't see security benefits of adding it.

Also, it would require a sender to track its own up-to-date endpoint information, which is not trivial (e.g. it may not know its own public IP outside NAT or it may not spot change of it, which happens in mobile networks).